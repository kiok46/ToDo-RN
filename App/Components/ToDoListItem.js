import React, { Component } from 'react';
import {
    Text,
    View,
    StyleSheet,
    TouchableOpacity
} from 'react-native';
import Card from './Card';

import PropTypes from 'prop-types';


class ToDoListItem extends Component {
	constructor(props) {
		super(props);
	}
	render (){
		return (
          <Card>
            <View style={styles.cardContainerStyle}>
              <View style={styles.contentTextContainerStyle}>
                <Text style={styles.contentTextStyle}>
                    {this.props.contentText}
                </Text>
              </View>
              <TouchableOpacity>
                <View>
                  <Text>Attach media</Text>
                </View>
              </TouchableOpacity>
            </View>
          </Card>
		);
	}
};

const styles = StyleSheet.create({
    cardContainerStyle: {
        flex: 1,
        justifyContent: 'space-around',
        height: 250,
        width: "100%",
        backgroundColor: 'rgba(230, 230, 230, 1)'
    },
    contentTextContainerStyle: {
        justifyContent: 'center',
        alignItems: 'center'
    },
    contentTextStyle: {
        fontWeight: '500',
        fontSize: 24
    }
});


ToDoListItem.defaultProps = {
    contentText: "",
}

ToDoListItem.propTypes = {
    contentText: PropTypes.string,
}

export default ToDoListItem;