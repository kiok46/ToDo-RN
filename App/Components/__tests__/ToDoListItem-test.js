import React from 'react';
import ToDoListItem from '../ToDoListItem';
import { shallow } from 'enzyme';


describe('testing ToDoListItem rendering', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <ToDoListItem />
        )
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ contentText: "Need to buy biscuits" })
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ endAt: "Never" })
        expect(wrapper).toMatchSnapshot();
    })
})
