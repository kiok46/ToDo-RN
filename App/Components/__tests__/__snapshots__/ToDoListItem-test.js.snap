// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ToDoListItem rendering renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToDoListItem
    contentText=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "backgroundColor": "rgba(230, 230, 230, 1)",
            "flex": 1,
            "height": 250,
            "justifyContent": "space-around",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "500",
              }
            }
          >
            
          </Text>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Attach media
            </Text>
          </View>
        </TouchableOpacity>
      </View>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              
            </Text>
          </View>,
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>
            </View>
          </TouchableOpacity>,
        ],
        "style": Object {
          "backgroundColor": "rgba(230, 230, 230, 1)",
          "flex": 1,
          "height": 250,
          "justifyContent": "space-around",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              
            </Text>,
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 24,
                "fontWeight": "500",
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>
            </View>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Attach media",
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": "Attach media",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "backgroundColor": "rgba(230, 230, 230, 1)",
              "flex": 1,
              "height": 250,
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              
            </Text>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>
            </View>
          </TouchableOpacity>
        </View>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "500",
                  }
                }
              >
                
              </Text>
            </View>,
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Attach media
                </Text>
              </View>
            </TouchableOpacity>,
          ],
          "style": Object {
            "backgroundColor": "rgba(230, 230, 230, 1)",
            "flex": 1,
            "height": 250,
            "justifyContent": "space-around",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "500",
                  }
                }
              >
                
              </Text>,
              "style": Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                },
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Attach media
                </Text>
              </View>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Attach media
                </Text>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Attach media",
                  "ellipsizeMode": "tail",
                },
                "ref": null,
                "rendered": "Attach media",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`testing ToDoListItem rendering renders as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToDoListItem
    contentText="Need to buy biscuits"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "backgroundColor": "rgba(230, 230, 230, 1)",
            "flex": 1,
            "height": 250,
            "justifyContent": "space-around",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "500",
              }
            }
          >
            Need to buy biscuits
          </Text>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Attach media
            </Text>
          </View>
        </TouchableOpacity>
      </View>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              Need to buy biscuits
            </Text>
          </View>,
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>
            </View>
          </TouchableOpacity>,
        ],
        "style": Object {
          "backgroundColor": "rgba(230, 230, 230, 1)",
          "flex": 1,
          "height": 250,
          "justifyContent": "space-around",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              Need to buy biscuits
            </Text>,
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Need to buy biscuits",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 24,
                "fontWeight": "500",
              },
            },
            "ref": null,
            "rendered": "Need to buy biscuits",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>
            </View>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Attach media",
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": "Attach media",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "backgroundColor": "rgba(230, 230, 230, 1)",
              "flex": 1,
              "height": 250,
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              Need to buy biscuits
            </Text>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Attach media
              </Text>
            </View>
          </TouchableOpacity>
        </View>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "500",
                  }
                }
              >
                Need to buy biscuits
              </Text>
            </View>,
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Attach media
                </Text>
              </View>
            </TouchableOpacity>,
          ],
          "style": Object {
            "backgroundColor": "rgba(230, 230, 230, 1)",
            "flex": 1,
            "height": 250,
            "justifyContent": "space-around",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "500",
                  }
                }
              >
                Need to buy biscuits
              </Text>,
              "style": Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Need to buy biscuits",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 24,
                  "fontWeight": "500",
                },
              },
              "ref": null,
              "rendered": "Need to buy biscuits",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Attach media
                </Text>
              </View>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Attach media
                </Text>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Attach media",
                  "ellipsizeMode": "tail",
                },
                "ref": null,
                "rendered": "Attach media",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
